xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -724,6 +739,134 @@ def test_publish_automatically_after_delete_unit(self, default_ms):
+    def test_xblock_create_event(self, default_ms):
+        """
+        Check that XBLOCK_CREATED event is sent when xblock is created.
+        """
+        self.initdb(default_ms)
+
+        event_receiver = Mock()
+        XBLOCK_CREATED.connect(event_receiver)
+
+        test_course = self.store.create_course('test_org', 'test_course', 'test_run', self.user_id)
+
+        # create sequential to test against
+        sequential = self.store.create_child(self.user_id, test_course.location, 'sequential', 'test_sequential')
+
+        event_receiver.assert_called()
+
+        assert event_receiver.call_args.kwargs['signal'] == XBLOCK_CREATED
+        assert event_receiver.call_args.kwargs['xblock_info'].usage_key == sequential.location
+        assert event_receiver.call_args.kwargs['xblock_info'].block_type == sequential.location.block_type
+        assert event_receiver.call_args.kwargs['xblock_info'].version.for_branch(None) == sequential.location
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_xblock_update_event(self, default_ms):
+        """
+        Check that XBLOCK_UPDATED event is sent when xblock is updated.
+        """
+        self.initdb(default_ms)
+
+        event_receiver = Mock()
+        XBLOCK_UPDATED.connect(event_receiver)
+
+        test_course = self.store.create_course('test_org', 'test_course', 'test_run', self.user_id)
+
+        # create sequential to test against
+        sequential = self.store.create_child(self.user_id, test_course.location, 'sequential', 'test_sequential')
+
+        # Change the xblock
+        sequential.display_name = 'Updated Display Name'
+        self.store.update_item(sequential, user_id=self.user_id)
+
+        event_receiver.assert_called()
+
+        assert event_receiver.call_args.kwargs['signal'] == XBLOCK_UPDATED
+        assert event_receiver.call_args.kwargs['xblock_info'].usage_key == sequential.location
+        assert event_receiver.call_args.kwargs['xblock_info'].block_type == sequential.location.block_type
+        assert event_receiver.call_args.kwargs['xblock_info'].version.for_branch(None) == sequential.location
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_xblock_publish_event(self, default_ms):
+        """
+        Check that XBLOCK_PUBLISHED event is sent when xblock is published.
+        """
+        self.initdb(default_ms)
+        event_receiver = Mock()
+        XBLOCK_PUBLISHED.connect(event_receiver)
+        test_course = self.store.create_course('test_org', 'test_course', 'test_run', self.user_id)
+        # create sequential and vertical to test against
+        sequential = self.store.create_child(self.user_id, test_course.location, 'sequential', 'test_sequential')
+        self.store.create_child(self.user_id, sequential.location, 'vertical', 'test_vertical')
+        # publish sequential changes
+        self.store.publish(sequential.location, self.user_id)
```

Comment:
Add blank line. We follow the structure proposed in https://3.basecamp.com/3966315/buckets/12732851/documents/2737536654. As a summary we divide the tests in three blocks. mock things, call things and assert things.